src/menus/GaExportMenu/GaExportMenu.js

```diff
@@ -0,0 +1,141 @@
+import Dialog from "../../components/Dialog";
+import ExportResolutionSelect from "../ExportMenu/ExportResolutionSelect";
+import { optionsA3, optionsA4, sizesByFormat } from "../../utils/exportUtils";
+import ExportButton from "../../components/ExportButton/ExportButton";
+import useExportSelection from "../../utils/useExportSelection";
+import gbLegends from "../../img/geltungsbereicheLegends";
+
+const SWISS_CENTER = [903000, 5899500];
+
+const useStyles = makeStyles(() => ({
+  dialogContainer: {
+    maxWidth: 500,
+  },
+  resSelect: {
+    width: 190,
+  },
+  mainBody: {
+    display: "flex",
+    flexDirection: "column",
+    gap: 25,
+  },
+  footer: {
+    padding: "10px 0",
+    display: "flex",
+    alignItems: "center",
+    justifyContent: "flex-end",
+  },
+  checkboxLabel: {
+    padding: "10px 0 10px 10px",
+  },
+  dialogRoot: {
+    position: "fixed !important",
+  },
+}));
+
+const options = [...optionsA3, ...optionsA4].filter((opt) =>
+  /^(2|3)$/.test(opt.resolution),
+);
+
+function GaExportMenu({ showModal, onClose }) {
+  const { t } = useTranslation();
+  const classes = useStyles();
+  const exportSelection = useExportSelection(options);
+  const zoom = useSelector((state) => state.map.zoom);
+  const center = useSelector((state) => state.map.center);
+  const layers = useSelector((state) => state.app.activeTopic)?.layers;
+  const [exportFullMap, setExportFullMap] = useState(false);
+  const visibleLayerName = useMemo(() => {
+    return layers.find((l) => !l.get("isBaseLayer") && l.visible)?.name;
+  }, [layers]);
+  const pdfSize = useMemo(() => {
+    return sizesByFormat[exportSelection?.format];
+  }, [exportSelection]);
+  const mapSize = useMemo(() => {
+    if (!exportSelection || !pdfSize) return null;
+    let scale = 1;
+    if (exportFullMap) {
+      scale = exportSelection.format === "a3" ? 1.95 : 2.25; // Magic scale factor to fit the map to pdf format
+    }
+    return [pdfSize[0] * scale, pdfSize[1] * scale];
+  }, [exportFullMap, exportSelection, pdfSize]);
+  const scaleLineConfig = useMemo(() => {
+    const getScaleLinePositionFunc = gbLegends.find(
+      (l) => l.name === visibleLayerName,
+    )?.getScaleLinePosition;
+    if (!exportSelection || !pdfSize || !getScaleLinePositionFunc) return null;
+    const scaleLinePosition = getScaleLinePositionFunc(
+      pdfSize[0],
+      exportSelection.resolution,
+    );
+    return scaleLinePosition;
+  }, [exportSelection, pdfSize, visibleLayerName]);
+
+  const exportZoom = useMemo(() => {
+    if (!exportSelection || !exportFullMap) return zoom;
+    return exportSelection.format === "a3" ? 9.3 : 9;
+  }, [exportFullMap, exportSelection, zoom]);
+
+  return (
+    showModal && (
+      <Dialog
+        isModal
+        name="ga-export-menu"
+        title={<span>{t("Karte als PDF exportieren")}</span>}
+        className={`tm-dialog-container ${classes.dialogContainer}`}
+        classes={{ root: classes.dialogRoot }}
```

Comment:
What if we use 2 makeStyles results here,   so here we could have class={{dialogClasses}}